pkg/resource/api.go

```diff
@@ -38,7 +37,8 @@ const (
 // An APIPatchingApplicator applies changes to an object by either creating or
 // patching it in a Kubernetes API server.
 type APIPatchingApplicator struct {
-	client client.Client
+	client      client.Client
+	optionalLog logging.Logger // can be nil
```

Comment:
There's a `logging.NopLogger` implementation that does nothing when called. Typically instead of an `if logger == nil` check we'd have `NewAPIPatchingApplicator` default to using `logging.NopLogger` such that log calls are discarded.

Reply:
this is intentional as diffing is quite expensive.